common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -188,10 +188,21 @@ private void captureCiMetadata() {
             envVariable("BITRISE_BUILD_NUMBER").ifPresent(value ->
                 buildScan.value("CI build number", value));
         }
+
+        if (isGoCD()) {
+            envVariable("GO_SERVER_URL").ifPresent(url ->
+                buildScan.link("GoCD", url));
```

Comment:
More useful would be the direct URL to the build but it's not provided as a direct env var, it could be built from other available env vars but those generated URLs could change in a future GoCD version...

Reply:
We already construct URLs for [TeamCity](https://github.com/gradle/gradle-enterprise-build-config-samples/blob/56e4ab082ffb8887323ac7d791e3aa5134da229a/common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java#L115) and [GitHub Actions](https://github.com/gradle/gradle-enterprise-build-config-samples/blob/56e4ab082ffb8887323ac7d791e3aa5134da229a/common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java#L156). Would the logic for GoCD be a lot more complex, or the end result possibly more susceptible to change? 

I'd try to be consistent with other CI vendors unless there's a compelling reason not to be.

Reply:
oh thanks I haven't noticed that, so I will do the same as GoCD URL seems simple enough:
`http://localhost:8153/go/tab/build/detail/<pipeline_name>/<pipeline_number>/<stage_name>/<stage_number>/<job_name>`

I don't know if those URLs will change anytime soon, my concern is that if they change we won't know be notified (until a user tell us :)), and it will be probably hard to fix for all CI versions with various URLs format.